src/lib/containers/SynapseHomepage.tsx

```diff
@@ -0,0 +1,463 @@
+import React, { useEffect, useState } from 'react'
+import { Button } from 'react-bootstrap'
+import { ReactComponent as HomepageGraphic } from '../assets/decorations/homepage-composite.svg'
+import { ReactComponent as OrganizeResearchGraphic } from '../assets/decorations/explore-research.svg'
+import { ReactComponent as GetCreditGraphic } from '../assets/decorations/certified-microscope.svg'
+import { ReactComponent as CollaborateGraphic } from '../assets/decorations/collaborate.svg'
+import { ReactComponent as DataSharingGraphic } from '../assets/decorations/data-justice.svg'
+import RLogo from '../assets/icons/RLogo'
+import Python from '../assets/icons/Python'
+import Terminal from '../assets/icons/Terminal'
+import Java from '../assets/icons/Java'
+import ProjectViewCarousel from './home_page/project_view_carousel/ProjectViewCarousel'
+import {
+  getEndpoint,
+  BackendDestinationEnum,
+} from '../utils/functions/getEndpoint'
+import { SynapseClient } from '../utils'
+
+export type SynapseHomepageProps = {
+  token?: string
+}
+
+export const SynapseHomepage: React.FunctionComponent<SynapseHomepageProps> = ({
+  token,
+}) => {
+  const LOGIN_LINK = `${getEndpoint(
+    BackendDestinationEnum.PORTAL_ENDPOINT,
+  )}#!LoginPlace:0`
+  const REGISTRATION_LINK = `${getEndpoint(
+    BackendDestinationEnum.PORTAL_ENDPOINT,
+  )}#!RegisterAccount:0`
+
+  // To update the carousel component, create a new version in the view. Then update it here.
+  const PROJECT_VIEW_ENTITY_ID = 'syn23593547.1'
+
+  const [dashboardLink, setDashboardLink] = useState('')
+
+  useEffect(() => {
+    const getDashboardLink = async () => {
+      if (token) {
+        const userProfile = await SynapseClient.getUserProfile(token)
+        setDashboardLink(
+          `${getEndpoint(BackendDestinationEnum.PORTAL_ENDPOINT)}#!Profile:${
```

Comment:
can replace id with `v`, then no need to make a web request, no need for useEffect